shell/src/main/java/alluxio/cli/validation/hdfs/SecureHdfsValidationTask.java

```diff
@@ -79,36 +86,72 @@ public SecureHdfsValidationTask(String process, String path, AlluxioConfiguratio
 
   @Override
   public String getName() {
-    return String.format("ValidateKerberosForSecureHdfs%s", mProcess.toUpperCase());
+    // Convert the first char to upper case
+    char first = Character.toUpperCase(mProcess.charAt(0));
+    return String.format("ValidateKerberosForSecureHdfs%s", first + mProcess.substring(1));
   }
 
   @Override
   public ValidationUtils.TaskResult validate(Map<String, String> optionsMap) {
-    if (shouldSkip()) {
+    if (!HdfsConfValidationTask.isHdfsScheme(mPath)) {
+      mMsg.append("Skip this check as the UFS is not HDFS.\n");
       return new ValidationUtils.TaskResult(ValidationUtils.State.SKIPPED, getName(),
               mMsg.toString(), mAdvice.toString());
     }
+
+    ValidationUtils.TaskResult loadConfig = loadHdfsConfig();
+    if (loadConfig.getState() != ValidationUtils.State.OK) {
+      return loadConfig;
+    }
+
+    // The state is OK when the HDFS is secured
+    ValidationUtils.TaskResult hdfsSecured = validateSecureHdfs();
+    if (hdfsSecured.getState() != ValidationUtils.State.OK) {
+      return hdfsSecured;
+    }
+
     return validatePrincipalLogin();
   }
 
-  protected boolean shouldSkip() {
-    if (!HdfsConfValidationTask.isHdfsScheme(mPath)) {
-      mMsg.append("Skip this check as the UFS is not HDFS.\n");
-      return true;
-    }
-    String principal = null;
-    if (mConf.isSet(mPrincipalProperty)) {
-      principal = mConf.get(mPrincipalProperty);
+  private ValidationUtils.TaskResult validateSecureHdfs() {
+    // Skipped if HDFS is not Kerberized
+    // Ref: https://docs.cloudera.com/documentation/enterprise/5-16-x/topics
+    // /cdh_sg_hadoop_security_enable.html
+    String hadoopAuthentication = mCoreConf.getOrDefault(HDFS_AUTHENTICATION_KEY, "");
+    boolean authenticationEnabled =
+            hadoopAuthentication.equalsIgnoreCase(HDFS_AUTHENTICATION_VALUE);
+    String hadoopAuthorization = mCoreConf.getOrDefault(HDFS_AUTHORIZATION_KEY, "");
+    boolean authorizationEnabled = hadoopAuthorization.equals(HDFS_AUTHORIZATION_VALUE);
+    if (!authenticationEnabled && !authorizationEnabled) {
+      mMsg.append("HDFS is not Kerberized. Skip this test.");
+      return new ValidationUtils.TaskResult(ValidationUtils.State.SKIPPED, getName(),
+              mMsg.toString(), mAdvice.toString());
     }
-    if (principal == null || principal.isEmpty()) {
-      mMsg.append(String.format("Skip validation for secure HDFS. %s is not specified.%n",
-          PRINCIPAL_MAP.get(mProcess).getName()));
-      return true;
+
+    // Issue an error if the secured HDFS is not configured properly
+    if (!authenticationEnabled || !authorizationEnabled) {
+      mMsg.append(String.format("Found inconsistent configuration for Hadoop security."
```

Comment:
These properties are not bound to be enabled at the same time, you can have authentication set to "simple" but have the authorization on.

Reply:
OK. Removed the authorization check as only authentication matters in this check